### PR TITLE
docs(nakama): Add docs for Nakama and Cardinal

### PR DESCRIPTION
**Pull request by jerargus**
## What is the purpose of the change

Add docs that describe Nakama features and how those features are used in the starter-game-template to make a general purpose Cardinal plugin for Nakama.

### CODE REVIEW

File: docs/pages/Cardinal/ecs.mdx
Comment (pyrofolium) [inline docs/pages/Cardinal/ecs.mdx:23]: Maybe add links where the user can go to explore more about the ECS pattern.

File: docs/pages/Nakama/plugin.mdx
Comment (pyrofolium) [inline docs/pages/Nakama/plugin.mdx:20]: Are these the new endpoints? I would personally still leave the old ones there as it's still usable. The new endpoints are sort of rushed while the old ones are more stable and hardened for now. Just my opinion. 

Also the /read/game/position has been replaced with /query/game/position.

**Review by pyrofolium — APPROVED** [review]
LGTM

File: docs/pages/Nakama/plugin.mdx
Comment (smsunarto) [inline docs/pages/Nakama/plugin.mdx:27]: Seems like the >>> <<< thing is causing syntax error, and i think you forgot to add the link to the markdown here

**Review by smsunarto — APPROVED** [review]
I recommend we pause this until the OpenAPI/swagger stuff is in because that's going to change the endpoints a lot

**Review by smsunarto — CHANGES_REQUESTED** [review]
I recommend we pause this until the OpenAPI/swagger stuff is in because that's going to change the endpoints a lot

**Review by namrapatel — APPROVED** [review]

**Comment by technicallyty** [comment]
title here should be `docs: nakama and cardinal` , this isn't necessarily a feature PR

File: docs/pages/Nakama/plugin.mdx
Comment (technicallyty) [inline docs/pages/Nakama/plugin.mdx:3]:
```suggestion
Go-based Nakama containers can be extended using shared objects (aka [plugins](https://pkg.go.dev/plugin)). The [starter-game-template](https://github.com/Argus-Labs/starter-game-template) leverages this shared object pattern to expand on Nakama to better interact with a Cardinal based backend.
```

File: docs/pages/Nakama/plugin.mdx
Comment (technicallyty) [inline docs/pages/Nakama/plugin.mdx:13]: maybe this is not important, but there aren't enough args to `NewReadType` here. maybe we can ellipsis to indicate there should be more after "position"

File: docs/pages/Nakama/plugin.mdx
Comment (technicallyty) [inline docs/pages/Nakama/plugin.mdx:21]: wdyt of explaining the differences of behavior of responses btwn transactions and reads? i.e. reads reply directly thru the RPC request, txs get streamed thru the match?

File: docs/pages/Nakama/plugin.mdx
Comment (technicallyty) [inline docs/pages/Nakama/plugin.mdx:33]: maybe for description here we can describe the type of private key: `ECDSA`

**Comment by pyrofolium** [comment]
One thing I just thought of @jerargus. Mage has certain commands that just don't work and aren't clear about what they do.

I was just informed that something like mage start is actually part of starter-game-template.

File: docs/pages/Nakama/plugin.mdx
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx:20]: I'll hold off on submitting this PR until the new endpoints are merged.

File: docs/pages/Nakama/plugin.mdx
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: I wasn't sure on the syntax. Hopefully these are fixed now.

File: docs/pages/Nakama/intro.mdx
Comment (jerargus) [reply docs/pages/Nakama/intro.mdx]: done

File: docs/pages/Nakama/plugin.mdx
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: I added a placeholder function name that isn't actually defined anywhere. Let me know if you think this should be "..." instead (or maybe a stubbed anonymous function).

File: docs/pages/Nakama/plugin.mdx
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: Done.

File: docs/pages/Nakama/plugin.mdx
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: done

File: docs/pages/Nakama/plugin.mdx
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: done

**Comment by technicallyty** [comment]
we do have some hackers @ interchain HH that might find this doc useful in some capacity. maybe we can merge as-is so they have something to read to better understand how/why we're using nakama, and revisit this later. @smsunarto

**Comment by jerargus** [comment]
> we do have some hackers @ interchain HH that might find this doc useful in some capacity. maybe we can merge as-is so they have something to read to better understand how/why we're using nakama, and revisit this later. @smsunarto 
> 

@technicallyty I added sections for both the current endpoints as well as the endpoints Brian is working on.

File: docs/pages/Cardinal/ecs.mdx
Comment (jerargus) [reply docs/pages/Cardinal/ecs.mdx:23]: This is out of scope for this PR (focusing on Nakama docs).

**Review by technicallyty — APPROVED** [review]

**Review by smsunarto — APPROVED** [review]

**Pull request merged**
